### DISCUSSION TITLE
Can't pin extensions via .plist file for macOSX

### DISCUSSION THREAD

**Discussion by MlooneyHV** [mozilla/policy-templates#1079] [General]
Can't figure out how to pin extensions via a plist file for enterprise firefox for macOSX

This is what I currently have:

![Screenshot 2024-03-12 at 11 50 59 AM](https://github.com/mozilla/policy-templates/assets/95653569/74922cd6-5e8d-49c7-b945-e55e564481a6)

**Comment by mkaply** [comment]
What shows up when you to to about:policies?

Do the other aspects of ExtensionSettings work? 

And has that extension been installed in that Firefox version before.

**Reply by MlooneyHV** [reply]
This shows up:

![Screenshot 2024-03-12 at 12 03 56 PM](https://github.com/mozilla/policy-templates/assets/95653569/a90d0f07-6645-4b74-93f2-96b18b39ba8d)

Just setting up a plist now to deploy to all our users to get Keeper Password Manager extension installed, not able to be uninstalled, and pinned.

Everything works minus the pinning of the extension

**Comment by mkaply** [comment]
That's so odd. Does it work on a new profile?

**Reply by MlooneyHV** [reply]
With a fresh install it does pin the extension. 

So I guess just won't work on existing installs?

Also, is there a way to force pin so that it can't be unpinned?

**Reply by mkaply** [reply]
Yeah, right now it's only new installs. 

We don't have a way to force the pin. The decision was made to set the initial and let the user choose.

**Comment by mkaply** [comment]
So I've filed a bug to fix it for reinstalls.

https://bugzilla.mozilla.org/show_bug.cgi?id=1885885
